server.go

```diff
@@ -797,6 +796,13 @@ func (s *Server) Users() (a []*User) {
 	return a
 }
 
+// UsersLen returns the number of users.
+func (s *Server) UsersLen() int {
```

Comment:
+1 on the lock. Wasn't thinking about that.

However, now that you've abstracted this functionality, wouldn't a better function name be something like `UsersCount()`? `len()` is a bit implementation-specific.